### PR TITLE
Tool Manager fix and improvement

### PR DESCRIPTION
**Pull request by meckzqz**
### Purpose of this PR

Removed presumably obsolete code resulting in the selection mode to be set to face selection both when launching the editor, and when entering or exiting play mode. In addition, the behaviour in ProBuilderToolManager's Dispose method where the selection mode is reset to object selection has been removed, allowing your selection mode to persist after entering and exiting play mode.

### CODE REVIEW

File: Editor/EditorCore/ProBuilderEditor.cs [hunk 1]
```diff
@@ -49,7 +49,6 @@ public sealed class ProBuilderEditor : EditorWindow, IHasCustomMenu
         static ProBuilderToolManager toolManager => s_Instance != null ? s_Instance.m_ToolManager : null;
         internal EditorToolbar toolbar => m_Toolbar; // used by unit tests
         static ProBuilderEditor s_Instance;
-        static Pref<SelectMode> s_LastActiveSelectMode = new Pref<SelectMode>("editor.lastActiveSelectMode", SelectMode.Face);
```
Comment (karl-) [inline Editor/EditorCore/ProBuilderEditor.cs:52]: Should add `"editor.lastActiveSelectMode"` to `PreferenceUpdater.DeleteObsoletePreferences` to clean up unused pref keys.

File: Editor/EditorCore/ProBuilderToolManager.cs [hunk 2]
```diff
@@ -95,8 +95,6 @@ public void Dispose()
             m_IsDisposed = true;
             GC.SuppressFinalize(this);
 
-            SetSelectMode(SelectMode.Object);
```
Comment (lopezt-unity) [inline Editor/EditorCore/ProBuilderToolManager.cs:98]: Not sure this should be removed.
It ensures that Select mode (and in more recent version, the EditorTools Context) are restored when closing the ProBuilder editor window, insuring to restore GameObject context for the tools.
It sure set the selectmode to object when entering playmode but this is then updating by the probuilder editor.
What do you think @karl-  ?

**Review by lopezt-unity — CHANGES_REQUESTED** [review]

File: Editor/EditorCore/ProBuilderEditor.cs (hunk 1 above)
Comment (meckzqz) [reply Editor/EditorCore/ProBuilderEditor.cs:52]: this change has now been made and pushed

File: Editor/EditorCore/ProBuilderToolManager.cs (hunk 2 above)
Comment (karl-) [reply Editor/EditorCore/ProBuilderToolManager.cs:99]: Yes I think this needs to remain here.

**Review by JoelFortin — APPROVED** [review]

**Comment by karl-** [comment]
Closing this for now, once PR feedback is addressed please feel free to re-open.

**Pull request closed without merging**